Source/CarthageKitTests/Fakes/FakeURLSession.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+final class FakeDownloadTask: NSURLSessionDownloadTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeDataTask: NSURLSessionDataTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeURLSession: NSURLSession {
```

Comment:
Let me take some time to look through this code and think about testing.

IMO mocking is a code smell. It works well enough for adding tests to existing code, but it's much better to structure code so that it requires little to no mocking. So I'm not against mocking, but I'd also prefer to find an alternative if possible.

But I _absolutely_ appreciate you spending time trying to refactor and test this code.